docs/source/examples/elasticsearch_outlet.rst

```diff
@@ -0,0 +1,85 @@
+        :end-at: raise RuntimeError(f"Index '{self.index_name}' does not exist ")
+
+    #. In this implementation of the :code:`push` method there are a few custom behaviors specified. As we iterate over every incoming record:
+
+        * We use the dict keys from the current record's :code:`payload` as our unique document id.
+        * The flag :code:`self.overwrite_documents` determines whether we will check if an id already exists. 
+        * If :code:`self.overwrite_documents` is True we simply index the document and :code:`_id` without doing any check. 
+        * Otherwise we use the client to check if :code:`_id` exists in the index. If it does we skip and log that it already exists. Otherwise it is indexed as normal. 
+
+    .. rst-class:: highlight-small
+    .. literalinclude:: ../../examples/elasticsearch_outlet.py
+        :language: python
+        :start-at: def push
+        :end-before: class DummyTextInlet
+
+    #. This simple :any:`Inlet` takes a list of strings as its main parameter. In it's :code:`pull` method it randomly selects one and returns the string and an incrementing id as a :code:`dict`. We'll use this to pass documents to our Elasticsearch Outlet. 
+
+    .. rst-class:: highlight-small
+    .. literalinclude:: ../../examples/elasticsearch_outlet.py
+        :language: python
+        :start-at: class DummyTextInlet
+        :end-at:   return {self._id: text_selection}
+
+    #. Instantiate our simple :any:`Inlet` as well as an instance of :code:`ElasticSearchIndexerOutlet` with the default parameter for :code:`overwrite_documents`.
+
+        * We use the official elasticsearch python client for `es_client`. 
+        * This example assumes `my-test-index` exists already in our elasticsearch cluster. 
+        * The link is setup to index a new document every 2 seconds. 
+    
+    .. rst-class:: highlight-small
+    .. literalinclude:: ../../examples/elasticsearch_outlet.py
+        :language: python
+        :start-at: es_client = elasticsearch.Elasticsearch(timeout=30)
+        :end-at:  planner.start
+
+    Output:
+
+    .. rst-class:: highlight-small
+    .. code-block:: python
+
+        >>> 2020-11-22 18:26:23.597|I| Indexed document with id 1 (databay.elasticsearch_outlet)
+        >>> 2020-11-22 18:26:25.498|I| Indexed document with id 2 (databay.elasticsearch_outlet)
+        >>> 2020-11-22 18:26:27.503|I| Indexed document with id 3 (databay.elasticsearch_outlet)
+        >>> 2020-11-22 18:26:29.497|I| Indexed document with id 4 (databay.elasticsearch_outlet)
+        >>> 2020-11-22 18:26:31.495|I| Indexed document with id 5 (databay.elasticsearch_outlet)
+        >>> 2020-11-22 18:26:33.492|I| Indexed document with id 6 (databay.elasticsearch_outlet)
+        >>> 2020-11-22 18:26:35.492|I| Indexed document with id 7 (databay.elasticsearch_outlet)
+        >>> 2020-11-22 18:26:37.493|I| Indexed document with id 8 (databay.elasticsearch_outlet)
+    
```

Comment:
I'd add a little explanation of the outputs here below the outputs themselves, even though it may seem unnecessary Something as simple as `We can see the records being written into our Elasticsearch index.` should do.

Also, for readability, I remove the logger's formatting from these example outputs - sorry I forgot to mention this yesterday. So this would read as:

```python
>>> Indexed document with id 1
>>> Indexed document with id 2
...
```